packages/fetchai/connections/http_client/connection.py

```diff
@@ -0,0 +1,276 @@
+            status_text=http_response.reason,
+            bodyy=http_response.content if http_response.content is not None else b"",
+            version="",
+        )
+        envelope = Envelope(
+            to=self.agent_address,
+            sender="HTTP server",
+            protocol_id=PublicId.from_str("fetchai/http:0.1.0"),
+            # context=context,
+            message=HttpSerializer().encode(http_message),
+        )
+        return envelope
+
+    def disconnect(self) -> None:
+        """
+        Disconnect.
+
+        Join the thread, then stop the channel.
+        """
+        if not self.is_stopped:
+            logger.info(f"HTTP Client has shutdown on port: {self.port}.")
+            self.is_stopped = True
+
+
+class HTTPClientConnection(Connection):
+    """Proxy to the functionality of the web RESTful API."""
+
+    def __init__(
+        self,
+        agent_address: Address,
+        provider_address: str,
+        provider_port: int,
+        *args,
+        **kwargs,
+    ):
+        """
+        Initialize a connection to a RESTful API.
+
+        :param agent_address: the address of the agent.
+        :param provider_address: RESTful API hostname / IP address
+        :param provider_port: RESTful API port number
+        """
+
+        if kwargs.get("connection_id") is None:
+            kwargs["connection_id"] = PublicId("fetchai", "http_client", "0.1.0")
+
+        super().__init__(*args, **kwargs)
+        self.agent_address = agent_address
+        self.channel = HTTPClientChannel(
+            agent_address,
+            provider_address,
+            provider_port,
+            excluded_protocols=self.excluded_protocols,
+            restricted_to_protocols=kwargs.get("restricted_to_protocols", {}),
+        )
+
+    async def connect(self) -> None:
+        """
+        Connect to a HTTP server.
+
+        :return: None
+        """
+        if not self.connection_status.is_connected:
+            self.connection_status.is_connected = True
+            self.channel.in_queue = asyncio.Queue()
+            self.channel.loop = self.loop
+            self.channel.connect()
+
+    async def disconnect(self) -> None:
+        """
+        Disconnect from a HTTP server.
+
+        :return: None
+        """
+        if self.connection_status.is_connected:
+            self.connection_status.is_connected = False
+            self.channel.disconnect()
+
+    async def send(self, envelope: "Envelope") -> None:
+        """
+        Send an envelope.
+
+        :param envelope: the envelop
+        :return: None
+        """
+        if not self.connection_status.is_connected:
+            raise ConnectionError(
+                "Connection not established yet. Please use 'connect()'."
+            )  # pragma: no cover
+        self.channel.send(envelope)
+
+    async def receive(self, *args, **kwargs) -> Optional["Envelope"]:
+        """
+        Receive an envelope.
```

Comment:
Note, this is an envelope the connection receives from the channel, so this is called by the multiplexer to get the latest envelopes from the HTTP client connection (i.e. the envelope with a http message of RESPONSE performative).